src/shop_gui.cpp

```diff
@@ -14,6 +14,58 @@
 #include "language.h"
 #include "sprite_store.h"
 #include "shop_type.h"
+#include "entity_gui.h"
+
+/** Window to prompt for removing a shop. */
+class ShopRemoveWindow : public EntityRemoveWindow  {
+	public:
+		ShopRemoveWindow(ShopInstance *si);
+
+		void OnClick(WidgetNumber number, const Point16 &pos) override;
+		void SetWidgetStringParameters(WidgetNumber wid_num) const override;
+
+	private:
+		ShopInstance *si; ///< Shop instance to remove.
+};
+
+/**
+ * Constructor of the shop remove window.
+ * @param si Shop instance to remove.
+ */
+ShopRemoveWindow::ShopRemoveWindow(ShopInstance *si) : EntityRemoveWindow(WC_SHOP_REMOVE, si->GetIndex())
+{
+	this->si = si;
+}
+
+void ShopRemoveWindow::OnClick(WidgetNumber number, const Point16 &pos)
+{
+	if (number == ERW_YES) {
+		Window *w = GetWindowByType(WC_SHOP_MANAGER, this->si->GetIndex());
+		delete w;
```

Comment:
Is it worth bothering with a variable here, if you're just going to delete it? (Or does the compiler require a variable? I'm not sure)